editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2599,6 +2599,11 @@ void CanvasItemEditor::_gui_input_viewport(const Ref<InputEvent> &p_event) {
 
 void CanvasItemEditor::_update_cursor() {
 	CursorShape c = CURSOR_ARROW;
+	bool should_switch = false;
+	if (drag_selection.size() != 0) {
+		float angle = drag_selection[0]->_edit_get_rotation();
+		should_switch = abs(Vector2(1, 0).rotated(angle).dot(Vector2(1, 0))) < abs(Vector2(1, 0).rotated(angle).dot(Vector2(0, 1)));
```

Comment:
We don't need to perform the dot product twice, the values on each side are inversely related, matching up at `Math_SQRT12`. So we can just compare with `Math_SQRT12`.

```suggestion
		should_switch = abs(Vector2(1, 0).rotated(angle).dot(Vector2(1, 0))) < Math_SQRT12;
```

Reply:
Oh yes, you are right. That's a good suggestion.